### ISSUE TITLE
MPRIS timer problem(s)

### ISSUE THREAD

**Issue by FichteFoll** [z411/trackma#420]
So, uhm, no particular idea what happened here, but this isn't right.

```
FichteFoll [anilist] (anime) watching >> update 21 7
Engine: Updating show Saki to episode 7...
[D] Data: Saving queue...
[D] Data: Saving cache...
Data: Queued update for Saki
[D] Data: Queued: my_progress -> 7
[D] Tracker (MPRIS): New playback status: Paused
[D] Tracker (MPRIS): New video: Saki - 08 - The Night Before
[D] Tracker (MPRIS): Guessing filename: Saki - 08 - The Night Before
Tracker (MPRIS): Will update Saki 8
[D] Tracker (MPRIS): (1) Setting active player: :1.7073
Data: Processing queue...
libanilist: Updating item Saki...
[D] Data: Saving cache...
[D] Data: Saving queue...
[D] Tracker (MPRIS): New playback status: Playing
[D] Data: No items in queue.
[D] Data: No items in queue.
[D] Data: No items in queue.
[D] Data: No items in queue.
[D] Tracker (MPRIS): New playback status: Paused
[D] Tracker (MPRIS): New playback status: Playing
[D] Tracker (MPRIS): New video: Saki - 09 - Eyes Open
[D] Tracker (MPRIS): Guessing filename: Saki - 09 - Eyes Open
Tracker (MPRIS) warning: Not playing the next episode of Saki. Ignoring.
…
FichteFoll [anilist] (anime) watching >> ls
| Index  Title    Progress  Score  |
…
|   21   Saki....   7 / 25    0    |
…
```

**Comment by z411** [comment]
> Tracker (MPRIS): Will update Saki 8

The Tracker was supposed to update to this episode, what happened when the timer got to 0?

**Comment by FichteFoll** [comment]
Nothing, I guess? My timer is 5 minutes and the episode was over 20 minutes long. I only noticed this after finishing the session at episode 12.

**Comment by z411** [comment]
I suspect the MPRIS timer is getting stuck (countdown doesn't start) under certain conditions. Will look into that.

**Comment by FichteFoll** [comment]
Whatever happened here…

```
[D] Tracker (MPRIS): New playback status: Playing
[D] Tracker (MPRIS): New video: [Underwater-FFF] Saki Achiga-hen - Episode of Side-A - 01 [BD][1080p-FLAC][69F59A59].mkv
[D] Tracker (MPRIS): Guessing filename: [Underwater-FFF] Saki Achiga-hen - Episode of Side-A - 01 [BD][1080p-FLAC][69F59A59].mkv
Tracker (MPRIS): Will update Saki: Achiga-hen - Episode of Side-A 1
[D] Tracker (MPRIS): (1) Setting active player: :1.26723
[D] Data: No items in queue.
[D] Data: No items in queue.
[D] Data: No items in queue.
[D] Data: No items in queue.
[D] Data: No items in queue.
[D] Tracker (MPRIS): New playback status: Paused
FichteFoll [anilist] (anime) watching >> tracker
- Tracker status -
State: Playing
Filename: [Underwater-FFF] Saki Achiga-hen - Episode of Side-A - 01 [BD][1080p-FLAC][69F59A59].mkv
Timer: 3453 [P]
Show: Saki: Achiga-hen - Episode of Side-A
Episode: 1
```

```
…
    "tracker_type": "mpris",
    "tracker_update_close": false,
    "tracker_update_prompt": false,
    "tracker_update_wait_s": 300,
…
```

**Comment by z411** [comment]
> Timer: 3453 [P]

Whoops... yeah that definitely seems related to the timer. I'll take a look tonight.

**Comment by FichteFoll** [comment]
Something's definitely wrong about the timer. It should never have more than 300s remaining, yet it starts with >600 this time.

```
[D] Tracker (MPRIS): New playback status: Playing
[D] Tracker (MPRIS): New video: [Underwater-FFF] Saki Achiga-hen - Episode of Side-A - 02 [BD][1080p-FLAC][BF12CE15].mkv
[D] Tracker (MPRIS): Guessing filename: [Underwater-FFF] Saki Achiga-hen - Episode of Side-A - 02 [BD][1080p-FLAC][BF12CE15].mkv
Tracker (MPRIS): Will update Saki: Achiga-hen - Episode of Side-A 2
[D] Tracker (MPRIS): (1) Setting active player: :1.30422
FichteFoll [anilist] (anime) watching >> tracker
- Tracker status -
State: Playing
Filename: [Underwater-FFF] Saki Achiga-hen - Episode of Side-A - 02 [BD][1080p-FLAC][BF12CE15].mkv
Timer: 654
Show: Saki: Achiga-hen - Episode of Side-A
Episode: 2
```

I watched exactly 2 episodes of a different show during this session before this log.

**Comment by z411** [comment]
I could reproduce it. The pause offset wasn't being cleared up. Please check if that fixes it as I'm not 100% sure it was the same issue.

**Comment by FichteFoll** [comment]
Something you did broke the "discard player with unknown shows" feature.

```
[D] Tracker (MPRIS): New playback status: Paused
[D] Tracker (MPRIS): New video: Boris Brejcha - DJ Mix Number 07 - 2018
[D] Tracker (MPRIS): Guessing filename: Boris Brejcha - DJ Mix Number 07 - 2018
[D] Tracker (MPRIS): Show guess: Boris Brejcha - DJ Mix Number: None
Tracker (MPRIS) warning: Found player but show not in list.
[D] Tracker (MPRIS): New tracker status: 3 (3)
[D] Tracker (MPRIS): (3) Setting active player: :1.940
[D] Tracker (MPRIS): Starting MPRIS timer.
```

All further *actual* episodes I watched after that were ignored due to originating from an ignored player.

**Comment by z411** [comment]
The log you posted is normal. The timer now gets started anyway since some users use the Prompt for Add function that requires a countdown even if the show isn't in the list.

Can you show the rest of the log and the exact steps to reproduce it? I tried opening an unrecognized video first then a proper episode and it gets recognized properly for me and not ignored.

```
Tracker (MPRIS): Connecting to MPRIS player: org.mpris.MediaPlayer2.mpv                   
[D] Tracker (MPRIS): New playback status: Playing                                         
[D] Tracker (MPRIS): New playback status: Playing                                         
[D] Tracker (MPRIS): New video: ASdwdjoiw.mp4                                             
[D] Tracker (MPRIS): Guessing filename: ASdwdjoiw.mp4                                     
[D] Tracker (MPRIS): Show guess: ASdwdjoiw: None                                          
Tracker (MPRIS) warning: Found player but show not in list.                               
[D] Tracker (MPRIS): New tracker status: 3 (3)                                            
[D] Tracker (MPRIS): (3) Setting active player: :1.66                                     
[D] Tracker (MPRIS): Starting MPRIS timer.                                                
[D] Tracker (MPRIS): New playback status: Playing                                         
[D] Tracker (MPRIS): New playback status: Playing                                         
[D] Tracker (MPRIS): New video: [Over-Time] Star☆Twinkle Precure - 02 [7393FF37].mkv      
[D] Tracker (MPRIS): Guessing filename: [Over-Time] Star☆Twinkle Precure - 02 [7393FF37].mkv
[D] Tracker (MPRIS): Show guess: Star☆Twinkle Precure: {'id': 105857, 'title': 'Star ☆ Twinkle Precure', 'my_progress': 1, 'total': 0, 'type': None, 'titles': ['Star ☆ Twinkle Precure', 'Star ☆ Twinkle Precure', 'スター☆トゥインクルプリキュア', 'Star ☆ Twinkle Pretty Cure', 'Star☆Twinkle Precure']}
Tracker (MPRIS): Will update Star ☆ Twinkle Precure 2                                     
[D] Tracker (MPRIS): New tracker status: 1 (1)                                            
[D] Tracker (MPRIS): (1) Setting active player: :1.66                                     
[D] Tracker (MPRIS): Starting MPRIS timer.                                                
[D] Tracker (MPRIS): New playback status: Playing                                         
z411 [anilist] (anime) watching >> tracker
- Tracker status -
State: Playing
Filename: [Over-Time] Star☆Twinkle Precure - 02 [7393FF37].mkv
Timer: 26
Show: Star ☆ Twinkle Precure
Episode: 2
```

**Comment by FichteFoll** [comment]
I don't have the rest of the log anymore, unfortunately, because of a power outage and losing the terminal's history because of that. However, I'm quite confident that the entire rest of the log consisted of a new player being found and then all further events being disregarded because the player was "ignored".

Maybe that was because the new player initially opened without a video and only had a video attached when playback was started? Syncplay starts the player without a video loaded, usually, and loads it only once it received the playlist or when you added something. (I know that I had started trackma after the video I included in the log and syncplay last.)
